components/org.wso2.carbon.identity.oidc.session/src/main/java/org/wso2/carbon/identity/oidc/session/servlet/OIDCLogoutServlet.java

```diff
@@ -431,14 +440,36 @@ private String extractClientFromIdToken(String idToken) throws ParseException {
     }
 
     /**
-     * Extract Subject from id token
+     * Extract tenant domain from id token
      * @param idToken id token
-     * @return Authenticated Subject
+     * @return tenant domain
      * @throws ParseException
      */
-    private String extractSubjectFromIdToken(String idToken) throws ParseException {
+    private String extractTenantDomainFromIdToken(String idToken) throws ParseException {
 
-        return SignedJWT.parse(idToken).getJWTClaimsSet().getSubject();
+        String tenantDomain = null;
+        Map realm = null;
+
+        JWTClaimsSet claimsSet = SignedJWT.parse(idToken).getJWTClaimsSet();
+        if (claimsSet.getClaims().get(OAuthConstants.OIDCClaims.REALM) instanceof Map) {
+            realm = (Map) claimsSet.getClaims().get(OAuthConstants.OIDCClaims.REALM);
+        }
+        if (realm != null) {
+            tenantDomain = (String) realm.get(OAuthConstants.OIDCClaims.TENANT);
+        }
+        if (StringUtils.isBlank(tenantDomain)) {
+            //It is not sending tenant domain with the subject in id_token by default, So to work this as
+            //expected, need to enable the option "Use tenant domain in local subject identifier" in SP config
+            tenantDomain = MultitenantUtils.getTenantDomain(claimsSet.getSubject());
+            if (log.isDebugEnabled()) {
+                log.debug("User tenant domain found in 'subject' claim of JWT. Tenant domain : " + tenantDomain);
```

Comment:
change found to derived & subject to 'sub'
add info on tried with realm but falling back to sub